cowprotocol/cowswap
# [EthFlow]#1294 wire is refunded

# Summary

Wiring up refund state when order expires

# To Test

To test this, you must make one order expire, which the only way I found is to move the price of the pair against you.

1. Pick ETH as sell
2. Make sure deadline and slippage are set to the minimum possible (10min and 2% respectively) to make your life easier
3. Pick as buy a token that you have enough funds to move the liquidity a few %
4. Fill in a significant ETH value, for example 7 ETH (I only tested on Goerli!)
5. Open another browser window (or another browser/device) and select the same pair
6. Set the sell amount to an amount similar but not identical - like 7.001 - to avoid the identical order issues
7. Place the order on both windows in close sucession
* One order should match and move the price enough so the other order becomes out of market
8. Wait until the pending order expires
* A few seconds after expiring (between 15 to 30) the order status will change from `refunding... etc` to `Refunded` like in this image
![Screenshot 2022-12-16 at 18 45 14](https://user-images.githubusercontent.com/43217/208167570-eaabd376-3834-4445-8f20-599af804c331.png)
